pkg/yamlmeta/node.go

```diff
@@ -48,32 +49,41 @@ func (ds *DocumentSet) SetValue(val interface{}) error {
 }
 
 func (d *Document) SetValue(val interface{}) error {
-	d.Value = val
-	return nil
+	d.ResetValue()
+	return d.AddValue(val)
 }
 
 func (m *Map) SetValue(val interface{}) error {
 	return fmt.Errorf("cannot set value on a map")
 }
 
 func (mi *MapItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into mapitem", val)
-	}
-	mi.Value = val
-	return nil
+	mi.ResetValue()
+	return mi.AddValue(val)
 }
 
 func (a *Array) SetValue(val interface{}) error {
 	return fmt.Errorf("cannot set value on an array")
 }
 
 func (ai *ArrayItem) SetValue(val interface{}) error {
-	if isMapOrArrayItem(val) {
-		return fmt.Errorf("cannot set map-or-array-item value (%T) into arrayitem", val)
+	ai.ResetValue()
+	return ai.AddValue(val)
+}
+
+func isValidValue(val interface{}) bool {
```

Comment:
looks like isMapOrArrayItem() is dead code after this.

Reply:
It is... I meant to delete but forgot. ty.